base/runtime/heap_allocator.odin

```diff
@@ -48,7 +51,7 @@ heap_allocator_proc :: proc(allocator_data: rawptr, mode: Allocator_Mode,
 		([^]rawptr)(aligned_mem)[-1] = allocated_mem
 
 		if force_copy {
-			mem_copy_non_overlapping(aligned_mem, old_ptr, old_size)
+			mem_copy_non_overlapping(aligned_mem, old_ptr, min(old_size, size))
```

Comment:
This corrupted heap when a resize happened that shrinks the original allocation, making the procedure copy more bytes than was left in the heap after OS resized our allocation.